internal/controller/csiaddons/persistentvolumeclaim_controller.go

```diff
@@ -856,3 +870,102 @@ func createAnnotationPredicate(annotations ...string) predicate.Funcs {
 		},
 	}
 }
+
+func (r *PersistentVolumeClaimReconciler) getScheduleFromSC(
+	ctx context.Context,
+	pvc *corev1.PersistentVolumeClaim,
+	logger *logr.Logger,
+	annotationKey string) string {
+	// For static provisioned PVs, StorageClassName is empty.
+	// Read SC schedule only when not statically provisioned.
+	if pvc.Spec.StorageClassName != nil && len(*pvc.Spec.StorageClassName) != 0 {
+		storageClassName := *pvc.Spec.StorageClassName
+		sc := &storagev1.StorageClass{}
+		err := r.Client.Get(ctx, types.NamespacedName{Name: storageClassName}, sc)
+		if err != nil {
+			if apierrors.IsNotFound(err) {
+				logger.Error(err, "StorageClass not found", "StorageClass", storageClassName)
+				return ""
+			}
+
+			logger.Error(err, "Failed to get StorageClass", "StorageClass", storageClassName)
+			return ""
```

Comment:
why we are returning empty in case of actual error, IMO this should be retried if there any problem in getting the SC

Reply:
`determineScheduleAndRequeue` will return the `ErrScheduleNotFound` upon getting an empty schedule from `getScheduleFromSC`

Reply:
what about the case where we failed to get the SC (apart from not found error)? we should retry in that case

Reply:
Done.